docs/index.md

```diff
@@ -0,0 +1,23 @@
+# ONS Technical Documentation
```

Comment:
Can you link to type hints doc here and then link this index doc from the readme please.